### ISSUE TITLE
Dynamic api

### ISSUE THREAD

**Issue by o1lo01ol1o** [Lysxia/type-map#2]
How should the empty `TypeMap x` be instantiated?  The constructor is not exposed from `.Internal`.  

Additionally, could the dynamic api support the `union`, `difference`, et al functions from `Data.Map` as currently implemented?

**Comment by Lysxia** [comment]
Thanks for bringing this up! It should be `empty`, and I forgot to export it, so importing `.Internal` is the best workaround for now. I only needed a small part of the library for my rewrite of boltzmann-samplers, and the rest (including all of `Dynamic`) was mainly a proof of concept for curiosity's sake. If that turns out to be useful, I'm definitely ready to expand the API!

A large part of `Data.Map` seems translatable to `type-map`.

PR and feature requests are both welcome.

**Comment by o1lo01ol1o** [comment]
That is what I thought was the case.  I ran into some issues when trying to create TypeMaps whose values depended on the types of the keys using the dynamic api.  Is this possible?  pseudo code:
```
foo = TM.insert (dummy :: Foo a) (MkConcrete (Foo 0)) TM.empty
foobar = TM.insert (dummy:: Bar a) (MkConcrete (Bar "Baz")) foo
```

Additionally, If you have any examples laying around they could help newcomers deduce the expected function of the library a little quicker.

**Comment by Lysxia** [comment]
[Here's a small demonstration of using `Data.TypeMap.Dynamic.Alt`](https://github.com/Lysxia/type-map/blob/9db11a78aa8b15088a9d1cce7349323b587bb2b6/examples/dynamic.hs).

I added the export of `empty` (https://github.com/Lysxia/type-map/commit/10da368be4b60505584147c5e84e1db07ad43ad5), and here are some more functions (https://github.com/Lysxia/type-map/commit/640b246705e23b98bb8cfd02753b81410cc04646).

**Comment by o1lo01ol1o** [comment]
Perfect, thanks.

**Comment by Lysxia** [comment]
I just released 0.1.2.0 containing the above changes!
